src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -2030,15 +2030,6 @@ void JavaThread::invoke_shutdown_hooks() {
     this->clear_pending_exception();
   }
```

Comment:
I think the HandleMark can be removed.
We may need to move lines 2026 - 2031 to the new `dump_at_exit` function.

Reply:
Thanks missed that. Not sure how we can get here with a pending exception though:
```
DestroyJavaVM()
  -> Threads::destroy_vm();
     -> wait till last non-daemon thread
     -> commit shutdown event
     -> JavaThread::invoke_shutdown_hooks() 
```
There is nowhere for an exception to materialize in this path AFAICS.

Reply:
Also note no need to move this as the code below clears any pending exception (from running the hooks). If we could get here with an exception pending it would have broken running of the hooks before we even had dynamic archiving.

Reply:
But looking at the code history we did always clear the exception before running the hooks. So I will restore this parts as it was before JDK-8266770. That means the HandleMark stays too.

Reply:
Thanks for the explanations.